### PR TITLE
DUPLO-15817 docs usage of cache_policy_id for cloudfront resource

### PR DESCRIPTION
**Pull request by Duplo-nikv**
## Overview

Updated document for cloudfront

## Summary of changes

This PR does the following:

- Updated docs
- Added description based on aws documentation

## Testing performed

- [ ] Using unit tests
- [ ] Manually, on my local system
- [ ] Manually, on a remote test system

## Describe any breaking changes

- ...

___

## **Type**
documentation, enhancement

___

## **Description**
- Added detailed field descriptions to the CloudFront distribution resource for better clarity on usage.
- Introduced documentation for the new `duplocloud_gke_credentials` data source.
- Updated the AWS CloudFront distribution, ECache instance, and GCP Node Pool resource documentation to reflect recent changes and corrections.
- Updated the changelog to include the latest documentation updates.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>resource_duplo_aws_cloudfront_distribution.go</strong><dd><code>Enhance CloudFront Distribution Resource with Detailed Field </code><br><code>Descriptions</code></dd></summary>
<hr>
      
duplocloud/resource_duplo_aws_cloudfront_distribution.go

<li>Added detailed descriptions for <code>cache_policy_id</code>, <code>default_ttl</code>, <code>headers</code>, <br><code>query_string_cache_keys</code>, <code>max_ttl</code>, and <code>min_ttl</code> fields.<br> <li> Enhanced documentation to include cache policy IDs and conditions when <br>certain fields are not required.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/446/files#diff-1627d8a34f08d738f48266f15aaabd490881def57ce69a352d25037768586085">+31/-17</a>&nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Documentation
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>CHANGELOG.md</strong><dd><code>Update Changelog for CloudFront Distribution Documentation Update</code></dd></summary>
<hr>
      
CHANGELOG.md

<li>Updated the changelog to include the documentation update for <br><code>duplocloud_aws_cloudfront_distribution</code> resource.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/446/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>gke_credentials.md</strong><dd><code>Add Documentation for GKE Credentials Data Source</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
docs/data-sources/gke_credentials.md

<li>Introduced a new data source documentation for <br><code>duplocloud_gke_credentials</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/446/files#diff-945c1b6054e2be42498cac013403d91967d2c979eaaa67602ae258e8f4d3c45a">+30/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>aws_cloudfront_distribution.md</strong><dd><code>Update AWS CloudFront Distribution Resource Documentation</code></dd></summary>
<hr>
      
docs/resources/aws_cloudfront_distribution.md

<li>Updated documentation to reflect the detailed descriptions added for <br>various fields in the CloudFront distribution resource.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/446/files#diff-dd6bb6b66c8f669f9f8731f37f92bf6e5e5abc93dbe29c0ee04201ed38fa6682">+14/-6</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ecache_instance.md</strong><dd><code>Update ECache Instance Documentation for Number of Shards</code></dd></summary>
<hr>
      
docs/resources/ecache_instance.md

<li>Modified the default value description for <code>number_of_shards</code> to be more <br>generic.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/446/files#diff-043ac248497c05f1e877b1714946cf707b361bbd70c813e754295d7750a58398">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>gcp_node_pool.md</strong><dd><code>Fix Typo in GCP Node Pool Resource Documentation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
docs/resources/gcp_node_pool.md

<li>Corrected the resource name from <code>duplocloud_gcp_k8_node_pool</code> to <br><code>duplocloud_gcp_k8_node_pools</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/446/files#diff-2adf3025f3b2f172abf5dd50661dc58fa6d0e2d8017f19036d05953de71f0b3b">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-15817 TF: docs usage of cache_policy_id for cloudfront resource](https://app.clickup.com/t/8655600/DUPLO-15817)

**Comment by Duplo-nikv** [comment]
Please refer documentation if I have covered all fields correctly
https://registry.terraform.io/providers/duplocloud/duplocloud/latest/docs/resources/aws_cloudfront_distribution

File: duplocloud/resource_duplo_aws_cloudfront_distribution.go [hunk 1]
```diff
@@ -438,16 +438,39 @@ func duploAwsCloudfrontDistributionSchema() map[string]*schema.Schema {
 						Type:     schema.TypeString,
 						Optional: true,
 						Computed: true,
+						Description: ` 
+						Amplify:
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_aws_cloudfront_distribution.go:442]: What is Amplify here, Can you please indent docs properly?

**Comment by Duplo-nikv** [comment]
policy id name

**Comment by sp1676** [comment]
/update_changelog --pr_update_changelog.push_changelog_changes=true

**Pull request merged**